dcpandas/df_builder.py

```diff
@@ -0,0 +1,292 @@
+            dates.remove('place')
+            row_max_date = max(dates)
+            if row_max_date > latest_date_so_far:
+                latest_date_so_far = row_max_date
+                latest_date = [options]
+            elif row_max_date == latest_date_so_far:
+                latest_date.append(options)
+    for options in most_geos:
+        if options in latest_date:
+            return rows_dict[options]
+
+
+def build_time_series_dataframe(places, stat_var, desc_col=False):
+    """Constructs a pandas DataFrame with `places` as the index and dates of the time series as the columns.
+
+    To ensure statistics are comparable across all Places, when multiple
+    StatVarObservations are available for Place and StatVar combos, Data
+    Commons selects the Observation options that covers the most Places, and breaks
+    ties using the Observation options that yield the latest Observation for any
+    Place.
+    
+    Args:
+      places (`str` or `iterable` of `str`): The dcids of Places to query for.
+      stat_var (`str`): The dcid of the StatisticalVariable.
+      desc_col: Whether to order columns in descending order.
+    Returns:
+      A pandas DataFrame with Place IDs as the index, and sorted dates as columns.
+    """
+    try:
+        if isinstance(places, six.string_types):
+            places = [places]
+        else:
+            places = list(places)
+            assert all(isinstance(place, six.string_types) for place in places)
+    except:
+        raise ValueError(
+            'Parameter `places` must be a string object or list-like object of string.'
+        )
+    if not isinstance(stat_var, six.string_types):
+        raise ValueError('Parameter `stat_var` must be a string.')
+
+    df = pd.DataFrame.from_records(_time_series_pd_input(places, stat_var))
+    df.set_index('place', inplace=True)
+    df.sort_index(inplace=True)
+    return df[sorted(df.columns, reverse=desc_col)]
+
+
+def _covariate_pd_input(places, stat_vars):
+    """Returns a `list` of `dict` per element of `places` based on the `stat_var`.
+
+    Data Commons will pick a set of StatVarObservation options that covers the
+    maximum number of queried places. Among ties, Data Commons selects an option
+    set with the latest Observation.
+
+    Args:
+      places (`str` or `iterable` of `str`): The dcids of Places to query for.
+      stat_vars (`Iterable` of `str`): The dcids of the StatisticalVariables.
+    Returns:
+      A `list` of `dict`, one per element of `places`. Each `dict` consists of
+      the time series and place identifier.
+
+    Examples:
+      >>> _covariate_pd_input(["geoId/29", "geoId/33"], ["Count_Person", "Median_Income_Person"])
+          [
+            {'Count_Person': 20, 'Median_Income_Person': 40, 'place': 'geoId/29'},
+            {'Count_Person': 428, 'Median_Income_Person': 429, 'place': 'geoId/33'}
+          ]
+    """
+
+    rows_dict = _group_stat_all_by_obs_options(places, stat_vars, 'covariates')
```

Comment:
for unit testing purposes, it's easier if this is passed in (rather than called from here)